promql/histogram_stats_iterator.go

```diff
@@ -19,32 +19,39 @@ import (
 	"github.com/prometheus/prometheus/tsdb/chunkenc"
 )
 
-type histogramStatsIterator struct {
+type HistogramStatsIterator struct {
 	chunkenc.Iterator
 
 	currentH *histogram.Histogram
 	lastH    *histogram.Histogram
 
 	currentFH *histogram.FloatHistogram
 	lastFH    *histogram.FloatHistogram
+
+	haveReadFromCurrentSeries bool
```

Comment:
Something rubs me the wrong way with this name. Maybe call it `currentSeriesRead`? Or `hasReadCurrentSeries`?

Different questions: With #16682 merged in, maybe this isn't required anymore?

Reply:
> Something rubs me the wrong way with this name. Maybe call it `currentSeriesRead`? Or `hasReadCurrentSeries`?

Changed in b88f19e.

> Different questions: With #16682 merged in, maybe this isn't required anymore?

See comment below - we need this or we lose some of the performance benefit of this change.

Reply:
Thanks. I didn't quite understand the combined implications with #16682.